zypp/parser/RepoindexFileReader.cc

```diff
@@ -245,9 +249,34 @@ namespace zypp
         if ( getAttrValue( "autorefresh", reader_r, attrValue ) )
           info.setAutorefresh( str::strToBool( attrValue, info.autorefresh() ) );
 
+        // optional *gpgcheck
+        if ( getAttrValue( "gpgcheck", reader_r, attrValue ) )
+          info.setGpgCheck( str::strToTriBool( attrValue ) );
+        if ( getAttrValue( "repo_gpgcheck", reader_r, attrValue ) )
+          info.setRepoGpgCheck( str::strToTrue( attrValue ) );
+        if ( getAttrValue( "pkg_gpgcheck", reader_r, attrValue ) )
+          info.setPkgGpgCheck( str::strToTrue( attrValue ) );
+
+        // optional keeppackages
+        if ( getAttrValue( "keeppackages", reader_r, attrValue ) )
+          info.setKeepPackages( str::strToTrue( attrValue ) );
+
+        // optional gpgkey
+        if ( getAttrValue( "gpgkey", reader_r, attrValue ) )
+          info.setGpgKeyUrl( Url(attrValue) );
+
+        // optional mirrorlist
+        if ( getAttrValue( "mirrorlist", reader_r, attrValue ) )
+          info.setMirrorListUrl( Url(attrValue) );
+
+        // optional metalink
+        if ( getAttrValue( "metalink", reader_r, attrValue ) )
+          info.setMetalinkUrl( Url(attrValue) );
```

Comment:
Should we support only one OR the other? IIRC RepoInfo would not support both... alternatively we could let the new  `effectiveBaseUrls` function in my PR just unite all the lists into one

Reply:
I followed the `ReopoInfoReader`. It also accepts both and sets both (first mirrorlist then metalink). `RepoInfo` internally uses just the last one set, so metalink superseeds mirrorlist. It's not different from specifying a baseurl in together with a metalink url. It's ok, the metalinkurl superseeds the baseurl.